src/components/tweet.js

```diff
@@ -44,15 +45,40 @@ export default class Tweet extends React.Component {
     return this.props.tweet.entities.media.map((media) => {
       if (media.type === 'photo') {
         return (
-          <a href={media.expanded_url} key={media.id_str} target='_blank'>
-            <img className='tweet_media' src={media.media_url} />
+          <a href="javascript:void(0);"  key={media.id_str} target='_blank'>
+            <img id ="tweet_mediaID"className='tweet_media' onClick={() => {this.openImageInWindow(media)}} src={media.media_url} />
           </a>
         );
       } else {
         return '';
       }
     });
   }
+  openImageInWindow (med) {
+    let image = new Image();
+    let loaded = false;
+    let mediaUrl = med.media_url;
+    let mediaWidth = 0;
+    let mediaHeight = 0;
+    image = document.getElementById('tweet_mediaID');
+    image.onload = function () { loaded = true; };
+
+    if (loaded) {
+        clearInterval(wait);
```

Comment:
No I have pushed any fixes yet. I have just edited the code so it's ready for pushing. I still need to figure out how to call my function from tweetContainer, as I don't code in Reactjs myself I have little understanding of its way of inheriting or referring to other components work?